src/definitions.ts

```diff
@@ -22,7 +22,7 @@ export enum Providers {
 }
 
 export interface Provider {
-  pin (hash: string, expectedSize: BigNumber): Promise<void>
+  pin (hash: string, expectedSize: BigNumber, agreementReference: string): Promise<void>
```

Comment:
Can we please don't pass Agreement Reference to the Provider but instead resolve it to PeerId in the processor and pass that? (as Provider is general, I would call it something like `directAddress`) Also, it should be optional parameter as there might not be the Peer Id...

I would like to keep these concepts separated for "separation of concerns".